master/buildbot/db/build_data.py

```diff
@@ -0,0 +1,167 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+import sqlalchemy as sa
+
+from twisted.internet import defer
+
+from buildbot.db import NULL
+from buildbot.db import base
+
+
+class BuildDataDict(dict):
+    pass
+
+
+class BuildDataConnectorComponent(base.DBConnectorComponent):
+
+    def _insert_race_hook(self, conn):
+        # called so tests can simulate a race condition during insertion
+        pass
+
+    @defer.inlineCallbacks
+    def setBuildData(self, buildid, name, value_json, source):
```

Comment:
I think buildData should be binary blob, and not only json. Encoding/decoding can be wherever is convenient. If this is json, responsible of the encode and decode could be the worker. transfer text between work may mean some encode decode in between, which can be avoided if we just use bytes everywhere until use.

Reply:
Agreed, changed to Blob/bytes